src/confluent_kafka/schema_registry/__init__.py

```diff
@@ -72,58 +72,82 @@
 ]
 
 
-def topic_subject_name_strategy(ctx, record_name: Optional[str]) -> Optional[str]:
+def topic_subject_name_strategy(ctx: Optional[SerializationContext], record_name: Optional[str]) -> Optional[str]:
     """
     Constructs a subject name in the form of {topic}-key|value.
 
     Args:
         ctx (SerializationContext): Metadata pertaining to the serialization
-            operation.
+            operation. **Required** - will raise ValueError if None.
 
         record_name (Optional[str]): Record name.
 
+    Raises:
+        ValueError: If ctx is None.
+
     """
+    if ctx is None:
```

Comment:
I refactored the name_strategy functions a bit: they have to follow the same function signature but SerializationContext is only required for some